app/script/cryptography/CryptographyMapper.js

```diff
@@ -127,22 +106,62 @@ z.cryptography.CryptographyMapper = class CryptographyMapper {
     };
   }
 
-  _map_image_asset_v3(asset, event_nonce) {
-    return {
-      data: {
-        content_length: asset.original.size.toNumber(),
-        content_type: asset.original.mime_type,
+  _map_asset(asset, event_nonce) {
+    const {original, preview, uploaded, not_uploaded} = asset;
+    let data = {};
+
+    if (original != null) {
+      data = {
+        content_length: original.size.toNumber(),
+        content_type: original.mime_type,
         info: {
-          height: asset.original.image.height,
+          name: original.name,
           nonce: event_nonce,
-          tag: 'medium',
-          width: asset.original.image.width,
         },
-        key: asset.uploaded.asset_id,
-        otr_key: new Uint8Array(asset.uploaded.otr_key.toArrayBuffer()),
-        sha256: new Uint8Array(asset.uploaded.sha256.toArrayBuffer()),
-        token: asset.uploaded.asset_token,
-      },
+      };
+
+      if (original.image) {
+        data.info.height = original.image.height;
+        data.info.width = original.image.width;
+      } else {
+        data.meta = this._map_asset_meta_data(original);
+      }
+    }
+
+    if (preview != null) {
```

Comment:
Sure its an Object (`not_uploaded` is the only exception). But since i want to check for existence, i think its still valuable to check in this way.